### ISSUE TITLE
INLINE parse error

### ISSUE THREAD

**Issue by Pitometsu** [ndmitchell/hlint#392]
`stack exec -- hlint` on module show:
```hs
Error: Parse error
Found:
    instance ToField UInt8 where
      toField = Plain . word8 . fromUInt8
  >             {-# INLINE toField #-}

    instance ToField UInt16 where


1 hint
```
And same error for `flycheck` in `emacs`.

While `stack build` is ok.

```bash
$ stack exec -- hlint --version
HLint v2.0.9, (C) Neil Mitchell 2006-2017
$ stack exec -- ghc --version
The Glorious Glasgow Haskell Compilation System, version 8.0.2
```

**Comment by Pitometsu** [comment]
With
```
instance ToField UInt8 where
  toField = Plain . word8 . fromUInt8
  {-# INLINE toField #-}
```
There's no error. Is previous formatting incorrect?

**Comment by ndmitchell** [comment]
I've never seen it written any way other than aligned as a statement... When I run the code through `ghc` I get:

```
Sample.hs:3:16: error: parse error on input `{-# INLINE'
  |
3 |                {-# INLINE toField #-}
  |                ^^^^^^^^^^
```

So looks invalid to my eye, and fails GHC as far as I can tell. Perhaps it is being conditionally ignored by GHC, but hlint isn't running the same CPP so it fails? Either way, it's highly unusual so I suggest fixing the code.
